test/datagenerator/1800medicare.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+type field struct {
+	l int
+	v string
+}
+
+type record struct {
+	hicn, blk                                    field
+	firstName, middleName, lastName              field
+	dob                                          field
+	addr1, addr2, addr3, city, state, zip5, zip4 field
+	gender                                       field
+	encDate                                      field
+	effDate, srcCode, mechCode, prefInd          field
+	saEffDate, saSrcCode, saMechCode, saPrefInd  field
+	acoID, acoName                               field
+}
+
+func main() {
+	rand.Seed(time.Now().UTC().UnixNano())
+	now := time.Now()
+
+	// 1-800-MEDICARE filename format: (T|P)#EFT.ON.ACO.NGD1800.DPRF.Dyymmdd.Thhmmsst
+	fileName := fmt.Sprintf("T#EFT.ON.ACO.NGD1800.DPRF.%s", now.Format("D060102.T1504050"))
+	outf, err := os.Create(fileName)
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = outf.WriteString(fmt.Sprintf("HDR_BENEDATASHR%s\n", now.Format("20060102")))
+	if err != nil {
+		panic(err)
+	}
+
+	for _, acoID := range []string{"A9990", "A9991", "A9992", "A9993", "A9994"} {
+		inf, err := os.Open(fmt.Sprintf("%s-HICNs", acoID))
+		if err != nil {
+			panic(err)
+		}
+
+		s := bufio.NewScanner(inf)
+
+		for s.Scan() {
+			// Randomly select HICNs from ACO for suppression records
+			if rand.Intn(2) == 0 {
+				continue
+			}
+
+			hicn := string(s.Bytes())
+			p := profile(hicn, acoID)
+			recs := records(p)
+			for _, r := range recs {
+				for _, f := range []field{r.hicn, r.blk, r.firstName, r.middleName, r.lastName, r.dob, r.addr1, r.addr2, r.addr3, r.city, r.state, r.zip5, r.zip4, r.gender, r.encDate, r.effDate, r.srcCode, r.mechCode, r.prefInd, r.saEffDate, r.saSrcCode, r.saMechCode, r.saPrefInd, r.acoID, r.acoName} {
+					_, err = outf.WriteString(fmt.Sprintf("%-"+fmt.Sprint(f.l)+"s", f.v))
+					if err != nil {
+						panic(err)
+					}
+				}
+				_, err = outf.WriteString("\n")
+				if err != nil {
+					panic(err)
+				}
+			}
+		}
+
+		err = inf.Close()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	err = outf.Close()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func profile(hicn string, acoID string) record {
+	dobMin, _ := time.Parse("2006-01-02", "1900-01-01")
+	dobMax := time.Now().Add(-65 * 365 * 24 * time.Hour)
+
+	p := record{
+		hicn:       field{l: 11, v: hicn},                     // HICN
+		blk:        field{l: 10},                              // Beneficiary link key
```

Comment:
is this blank? it's cool because that's valid just wanna make sure im reading this correctly

Reply:
Yes, any case where `v` isn't set will default to an empty string. If you think the bene link key would be useful, I'm happy to add it!

Reply:
nah I don't even know what it is :)